pkg/tasks/update_template_content.go

```diff
@@ -0,0 +1,655 @@
+package tasks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	caliri "github.com/content-services/caliri/release/v4"
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/candlepin_client"
+	"github.com/content-services/content-sources-backend/pkg/config"
+	"github.com/content-services/content-sources-backend/pkg/dao"
+	"github.com/content-services/content-sources-backend/pkg/db"
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"github.com/content-services/content-sources-backend/pkg/pulp_client"
+	"github.com/content-services/content-sources-backend/pkg/tasks/payloads"
+	"github.com/content-services/content-sources-backend/pkg/tasks/queue"
+	zest "github.com/content-services/zest/release/v2024"
+	"github.com/google/uuid"
+	"github.com/labstack/gommon/random"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"golang.org/x/exp/slices"
+)
+
+func UpdateTemplateContentHandler(ctx context.Context, task *models.TaskInfo, queue *queue.Queue) error {
+	opts := payloads.UpdateTemplateContentPayload{}
+	if err := json.Unmarshal(task.Payload, &opts); err != nil {
+		return fmt.Errorf("payload incorrect type for UpdateTemplateDistributions")
+	}
+
+	logger := LogForTask(task.Id.String(), task.Typename, task.RequestID)
+	ctxWithLogger := logger.WithContext(ctx)
+
+	daoReg := dao.GetDaoRegistry(db.DB)
+	domainName, err := daoReg.Domain.Fetch(ctxWithLogger, task.OrgId)
+	if err != nil {
+		return err
+	}
+
+	rhDomainName, err := daoReg.Domain.Fetch(ctxWithLogger, config.RedHatOrg)
+	if err != nil {
+		return err
+	}
+
+	pulpClient := pulp_client.GetPulpClientWithDomain(domainName)
+	cpClient := candlepin_client.NewCandlepinClient()
+
+	t := UpdateTemplateContent{
+		orgId:          task.OrgId,
+		ownerKey:       candlepin_client.DevelOrgKey,
```

Comment:
We should only use DevelOrgKey if its set (which shouldn't be for stage and prod), for those environments I *think* we can use the org_id as the owner key, but we may need to ask around about that.

Reply:
What do we do for now if it's not set? Right now the CreateOwner method errors if DevelOrg is not set.

Reply:
i suspect that the org_id we're storing and getting in the identity header is the owner 'key'.

Reply:
so do we unblock the owner creation?

Reply:
Oh actually, this doesn't call CreateOwner(). Nevermind.